projects/auth0-angular/src/lib/auth.service.ts

```diff
@@ -297,8 +347,12 @@ export class AuthService<TAppState extends AppState = AppState> implements OnDes
    *
    * @param url The URL to that should be used to retrieve the `state` and `code` values. Defaults to `window.location.href` if not given.
    */
-  handleRedirectCallback(url?: string): Observable<RedirectLoginResult<TAppState>> {
-    return defer(() => this.auth0Client.handleRedirectCallback<TAppState>(url)).pipe(
+  handleRedirectCallback(
+    url?: string
+  ): Observable<RedirectLoginResult<TAppState>> {
```

Comment:
Similar curiosity as for `getUser`, where `handleRedirectCallback` in SPA SDK allows for the type param to optionally be specified:

```js
public async handleRedirectCallback<TAppState = any>
```

I know this is not new code here, but thought I'd mention it.

Reply:
This works a bit different in the AngularService implementation:

```ts
const service = new AuthService<FooAppState>();
service.handleRedirectCallback().subscribe(result => {
  result.appState //<= Type FooAppState
});
```

Because we define the AppState type on the service, we do not need to set it on the `handleRedirectCallback`.
It could be a bit confusing, as this is different from the underlying SPA's. Could always change this in a backwards compatible way.

But regardless, type information should be correct regarding the AppState.